google-cloud-storage/test/google/cloud/storage/file_update_test.rb

```diff
@@ -106,6 +106,24 @@
     mock.verify
   end
 
+  it "updates its storage_class" do
+    mock = Minitest::Mock.new
+    patch_file_gapi = Google::Apis::StorageV1::Object.new storage_class: "DURABLE_REDUCED_AVAILABILITY"
+    patched_file_gapi = file_gapi.dup
+    patched_file_gapi.storage_class = "DURABLE_REDUCED_AVAILABILITY"
+    rewrite_resp = OpenStruct.new done: true, resource: patched_file_gapi
+    mock.expect :rewrite_object, rewrite_resp,
+      [bucket_name, file.name, bucket_name, file.name, patch_file_gapi]
+
+    file.service.mocked_service = mock
+
+    file.storage_class.must_equal "STANDARD"
+    file.storage_class = :dra
+    file.storage_class.must_equal "DURABLE_REDUCED_AVAILABILITY"
```

Comment:
The reason I chose it was because of the automatic mapping we do for `storage_class`. The values `:dra` and `"DURABLE_REDUCED_AVAILABILITY"` have the biggest difference, so I wanted the tests to exercise show that happens.